src/message_pool.zig

```diff
@@ -65,6 +66,34 @@ comptime {
 /// initialization and reused thereafter. The messages_max values determine the size of this pool.
 pub const MessagePool = struct {
     pub const Message = struct {
+        pub const Reserved = MessageType(.reserved);
+        pub const Ping = MessageType(.ping);
+        pub const Pong = MessageType(.pong);
+        pub const PingClient = MessageType(.ping_client);
+        pub const PongClient = MessageType(.pong_client);
+        pub const Request = MessageType(.request);
+        pub const Prepare = MessageType(.prepare);
+        pub const PrepareOk = MessageType(.prepare_ok);
+        pub const Reply = MessageType(.reply);
+        pub const Commit = MessageType(.commit);
+        pub const StartViewChange = MessageType(.start_view_change);
+        pub const DoViewChange = MessageType(.do_view_change);
+        pub const StartView = MessageType(.start_view);
+        pub const RequestStartView = MessageType(.request_start_view);
+        pub const RequestHeaders = MessageType(.request_headers);
+        pub const RequestPrepare = MessageType(.request_prepare);
+        pub const RequestReply = MessageType(.request_reply);
+        pub const Headers = MessageType(.headers);
+        pub const Eviction = MessageType(.eviction);
+        pub const RequestBlocks = MessageType(.request_blocks);
+        pub const Block = MessageType(.block);
+        pub const RequestSyncCheckpoint = MessageType(.request_sync_checkpoint);
+        pub const SyncCheckpoint = MessageType(.sync_checkpoint);
+        pub const RequestSyncFreeSet = MessageType(.request_sync_free_set);
+        pub const RequestSyncClientSessions = MessageType(.request_sync_client_sessions);
+        pub const SyncFreeSet = MessageType(.sync_free_set);
+        pub const SyncClientSessions = MessageType(.sync_client_sessions);
+
         // TODO: replace this with a header() function to save memory
```

Comment:
This is mostly unrelated to the PR, but I think there's a better solution here, which also gets rid of an extra level of indirection:

```zig
pub const MessageRc = extern struct align(constant.sector_size) {
    padding: [constant.sector_size - @sizeOf(u32) - @sizeOf(?*Message)]u8 = undefined,
    references: u32 = 0,
    next: ?*Message,
    message: Message,
};

pub const Message = extern struct {
    header: Header,
    body: [constants.message_size_max - @sizeOf(Header)]u8,
};
```

We'll still be passing around a `*Message`, but it'll now be a pointer directly into message bytes (as opposed to a pointer to  a pointer to a body). `message.header` and `message.body` would be direct references into the underlying memory (reducing the need to slice by `sizeOf(Header)`), and `mem.asBytes(message)` would give the raw memory of the entire message. 

The reference count and next link would be accessed through `@fieldParentPtr`. We'll lose some memory for left-padding the rc wrapper to sector boundary, but overall it feels like it should be a win, as we have less indirection.

Reply:
(alternatively, we could just tack `references` and `next` to the end of the message, after the body)

Reply:
> TODO: replace this with a header() function to save memory

Doesn't the `padding` field in `MessageRc` undo any memory won from not needing the `header` pointer in `Message`? I don't understand what this design would buy us.

Reply:
Yeah, the improvement here is not in saving memory, but in:

* removing the indirection (right now, `message.header.replica` is two pointer dereferences, but it could be one),
* simplifying the call-site, as `.body` is a field access, rather than a function call,

It will also:

* maaybe improving const-correcness, as `*const Message` would mean "immutable message", rather than "mutable message with immutable reference count" we have today. 
* reduce the _number_ of allocations, as we no longer need to allocate rc+pointers in a separate (tiny) allocation.